### PR TITLE
Cleaned up database functions

### PR DESCRIPTION
**Pull request by atmiguel**
As well as making a change to a handler function that evan recommended

### CODE REVIEW

File: databaseutil/databaseutil.go [hunk 1]
```diff
@@ -60,36 +66,58 @@ func InsertIntoUsersTable(
...
+	rows, err := db.Query(sqlQuery, emailAddress)
+	if err != nil {
+		return 0, convertPostgresError(err)
 	}
+	defer rows.Close()
 
 	var userId int64
-	if err := row.Scan(&userId); err != nil {
-		return 0, err
+	for rows.Next() {
```
Comment (zingales) [inline databaseutil/databaseutil.go:109]: The next time we do this exact logic we should make it its own separate function.

File: handlers/handlers.go [hunk 2]
```diff
@@ -334,15 +333,14 @@ func HandleNotesPageRequest(
 
 func respondWithMethodNotAllowed(
 	responseWriter http.ResponseWriter,
-	allowedMethods ...string,
+	allowedMethod string,
```
Comment (zingales) [inline handlers/handlers.go:336]: this is to guarantee there is at least one method, at compile time?

Comment (atmiguel) [reply handlers/handlers.go:336]: yeah

File: databaseutil/databaseutil.go (hunk 1 above)
Comment (atmiguel) [reply databaseutil/databaseutil.go:109]: agreed

**Review by zingales — APPROVED** [review]

**Review by zingales — APPROVED** [review]

**Pull request merged**